core/app/lib/uffizzi_core/concerns/models/cluster.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module UffizziCore::Concerns::Models::Cluster
+  extend ActiveSupport::Concern
+  include UffizziCore::ClusterRepo
+
+  included do
+    include AASM
+
+    self.table_name = UffizziCore.table_names[:clusters]
+
+    belongs_to :project, class_name: UffizziCore::Project.name
+    belongs_to :deployed_by, class_name: UffizziCore::User.name, foreign_key: :deployed_by_id, optional: true
+    validates_uniqueness_of :name, conditions: -> { enabled }
+    validates :name, presence: true, format: { with: /([A-Za-z0-9\-_]+)/ }
+
+    aasm(:state) do
+      state :deploying_namespace, initial: true
```

Comment:
If a user creates an empty cluster, will it be stuck in the `deploying_namespace` state?